drivers/serial/uart_bm.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for UART port on Bristlemouth
+ *
+ */
+
+#ifndef ZEPHYR_DRIVERS_SERIAL_UART_BM_H_
+#define ZEPHYR_DRIVERS_SERIAL_UART_BM_H_
+
+#include <drivers/pinctrl.h>
+
+#include <stm32_ll_usart.h>
+
+/* device config */
+struct uart_bm_config {
+	/* USART instance */
+	USART_TypeDef *usart;
+	/* clock subsystem driving this peripheral */
+	struct stm32_pclken pclken;
+	/* initial hardware flow control, 1 for RTS/CTS */
+	bool hw_flow_control;
+	/* initial parity, 0 for none, 1 for odd, 2 for even */
+	int  parity;
+	/* switch to enable single wire / half duplex feature */
+	bool single_wire;
+	const struct pinctrl_dev_config *pcfg;
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN) || defined(CONFIG_UART_ASYNC_API) || \
+	defined(CONFIG_PM)
+	uart_irq_config_func_t irq_config_func;
+#endif
+};
+
+#ifdef CONFIG_UART_ASYNC_API
+struct uart_dma_stream {
```

Comment:
This should probably be renamed to uart_bm_dma_stream, to prevent potential conflicts with the original uart_stm32 driver.